backend/data_validation/validation.js

```diff
@@ -0,0 +1,12 @@
+async function validateDeviceData(checkBody, validationErrors, body) {
+    try {
+        checkBody('chip_id').notEmpty().withMessage("required city");
+        checkBody('user_id').notEmpty()
+        let errors = validationErrors();
+        return body
+    } catch (err) {
+        res.status(400).json({ error: 1, message: "error during validating the data" });
```

Comment:
how are you getting res object here?

Reply:
sir, this is wrong, i need return error from this function then i will handle that at calling time